docs/user/pings/metrics.md

```diff
@@ -1,29 +1,36 @@
+The `metrics` ping is intended for all of the metrics that are explicitly set by the application or are included in the application's `metrics.yaml` file (except events). 
+The reported data is tied to the ping's *measurement window*, which is the time between the collection of two `metrics` ping. 
+Ideally, this window is expected to be about 24 hours, given that the collection is scheduled daily at 4AM. 
+However, the metrics ping is only sent while the application is actually running, so in practice, it may not meet the 4AM target very frequently.
+Data in the [`ping_info`](index.md#the-ping_info-section) section of the ping can be used to infer the length of this window and the reason that triggered the ping to be sent.
 If the application crashes, unsent recorded metrics are sent along with the next `metrics` ping.
 
+Additionally, it is undesirable to mix metric recording from different versions of the application. Therefore, if a version upgrade is detected, the `metrics` ping is collected immediately before further metrics from the new version are recorded.
+
 > **Note:** As the `metrics` ping was specifically designed for mobile operating systems, it is not sent when using the Glean Python bindings.
 
 ## Scheduling
-The desired behavior is to collect the ping at the first available opportunity after 4AM local time on a new calendar day.
+The desired behavior is to collect the ping at the first available opportunity after 4AM local time on a new calendar day, but given constraints of the platform, it can only be sent while the application is running. 
 This breaks down into three scenarios:
 
 1. the application was just installed;
-2. the application was just started (after a crash or a long inactivity period);
-3. the application was open and the 4AM due time was hit.
+2. the application was just upgraded (the version of the app is different from the last time the app was run);
+3. the application was just started (after a crash or a long inactivity period);
+4. the application was running at 4AM.
+
+In the first case, since the application was just installed, if the due time for the current calendar day has passed, a `metrics` ping is immediately generated and scheduled for sending (reason code `overdue`). Otherwise, if the due time for the current calendar day has not passed, a ping collection is scheduled for that time (reason code `today`). 
 
-In the first case, since the application was just installed, if the due time for the current calendar day has passed, a `metrics` ping is immediately generated and scheduled for sending. Otherwise, if the due time for the current calendar day has not passed, a ping collection is scheduled for that time.
+In the second case, if a version change is detected at startup, the metrics ping is immediately sent so that metrics from one version are not aggregated with metrics from another version (reason code `upgrade`).
```

Comment:
```suggestion
In the second case, if a version change is detected at startup, the metrics ping is immediately submitted so that metrics from one version are not aggregated with metrics from another version (reason code `upgrade`).
```